client.go

```diff
@@ -214,11 +214,10 @@ func (c *client) unassignRoles(ctx context.Context, roleIDs []string) error {
 	var merr *multierror.Error
 
 	for _, id := range roleIDs {
-		var rawResponse *http.Response
-		ctxWithResp := policy.WithCaptureResponse(ctx, &rawResponse)
-		if _, err := c.provider.DeleteRoleAssignmentByID(ctxWithResp, id); err != nil {
+		detailedErr := new(azcore.ResponseError)
+		if _, err := c.provider.DeleteRoleAssignmentByID(ctx, id); err != nil {
```

Comment:
nit: we only need to create the response err in the case where an error has occurred. We can probably rename the var to `respError` since there appears to be no other class of error returned by the deletion call.
```suggestion
		if _, err := c.provider.DeleteRoleAssignmentByID(ctx, id); err != nil {
				detailedErr := new(azcore.ResponseError)
```